packages-ts/gauntlet-terra-cw20-multisig/src/commands/multisig.ts

```diff
@@ -0,0 +1,250 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { logger, prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { TerraCommand, TransactionResponse } from '@chainlink/gauntlet-terra'
+import { AccAddress, MsgExecuteContract } from '@terra-money/terra.js'
+import { isDeepEqual } from '../lib/utils'
+
+type ProposalAction = (
+  signer: AccAddress,
+  proposalId: number,
+  message: MsgExecuteContract,
+) => Promise<MsgExecuteContract>
+
+enum Vote {
+  YES = 'yes',
+  NO = 'no',
+  ABS = 'abstain',
+  VETO = 'veto',
+}
+
+type WasmMsg = {
+  wasm: {
+    execute: {
+      contract_addr: string
+      funds: {
+        denom: string
+        amount: string
+      }[]
+      msg: string
+    }
+  }
+}
+
+enum Action {
+  CREATE = 'create',
+  APPROVE = 'approve',
+  EXECUTE = 'execute',
+  NONE = 'none',
+}
+
+type State = {
+  threshold: number
+  nextAction: Action
+  owners: AccAddress[]
+  approvers: string[]
+  // https://github.com/CosmWasm/cw-plus/blob/82138f9484e538913f7faf78bc292fb14407aae8/packages/cw3/src/query.rs#L75
+  currentStatus?: 'pending' | 'open' | 'rejected' | 'passed' | 'executed'
+  data?: WasmMsg[]
+}
+
+export const wrapCommand = (command) => {
+  return class Multisig extends TerraCommand {
+    command: TerraCommand
+    multisig: AccAddress
+    multisigGroup: AccAddress
+
+    static id = `${command.id}:multisig`
+
+    constructor(flags, args) {
+      super(flags, args)
+
+      this.command = new command(flags, args)
+
+      if (!AccAddress.validate(process.env.MULTISIG_ADDRESS)) throw new Error(`Invalid Multisig wallet address`)
+      if (!AccAddress.validate(process.env.MULTISIG_GROUP)) throw new Error(`Invalid Multisig group address`)
+      this.multisig = process.env.MULTISIG_ADDRESS as AccAddress
+      this.multisigGroup = process.env.MULTISIG_GROUP as AccAddress
```

Comment:
This should really be read from the input, and not directly from env vars here. Env vars should be completely abstracted from the command. The command just gets some kind of input (flags) that can be loaded from anywhere (CLI flags/args, env, file ...)

Reply:
Regarding loading environment variables, how about something like this:

```
const environment: Environment = load(.env filepath)

...

executeCLI(environment, myCommands)
```

Then, in each command:

```
this.multisig = this.env.MULTISIG_ADDRESS
```

where `this.env` contains the environment variables, available in each `Command` instance.

Reply:
Opening an issue here, thought could be a simpler fix, but Gauntlet is not really prepared to inject env vars anywhere.
https://github.com/smartcontractkit/chainlink-terra/issues/134